main.cpp

```diff
@@ -964,21 +188,21 @@ int main(int argc, char ** argv) {
 //    params.prompt = R"(// this function checks if the number n is prime
 //bool is_prime(int n) {)";
 
-    int64_t t_load_us = 0;
-
-    llama_vocab vocab;
-    llama_model model;
+    llama_context * ctx;
 
     // load the model
     {
-        const ggml_type memory_type = params.memory_f16 ? GGML_TYPE_F16 : GGML_TYPE_F32;
-        const int64_t t_start_us = ggml_time_us();
-        if (!llama_model_load(params.model, model, vocab, params.n_ctx, params.n_parts, memory_type)) {
-            fprintf(stderr, "%s: failed to load model from '%s'\n", __func__, params.model.c_str());
+        auto lparams = llama_context_default_params();
+
+        lparams.f16_kv = params.memory_f16;
+        lparams.logits_all = params.perplexity;
```

Comment:
missing parameters:
```
        lparams.n_parts = params.n_parts;
        lparams.n_ctx = params.n_ctx;
```

Reply:
They are initialized with defaults from the `llama_context_default_params()` call above

Reply:
Not copying these values from `params` breaks their respective command line switches (--n_parts and -c).

Reply:
Oops - missed that!